app/Family.php

```diff
@@ -206,21 +212,48 @@ public function getEligibleChildrenCountAttribute()
         });
     }
 
+
+    /**
+     * Generates and sets the components required for an RVID.
+     *
+     * @param Centre $centre
+     */
+    public function generateRVID(Centre $centre)
+    {
+        if (is_null($this->centre_sequence)) {
+            if ($centre) {
+                // Get the last family made in this centre.
+                $family = Family::where('initial_centre_id', $centre->id)
```

Comment:
Do we have soft Deletes on family - should we try to get trashed ones?

Reply:
Nope. not yet.